[Liew Jian Hong] iP

jianh0ng opened the pull request:
# Duke
Duke keeps tracks of the things you have to do, key features are: 
* text based
* simple commands
* efficient

How to use:
1. Choose 1 of 3 types of tasks, Todos, Deadlines and Events
2. Add the task in
3. Mark it as done after completion or delete it

Features
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Source code can be found [here](https://github.com/jianh0ng/ip)

Duke uses Java, so you can use it to practice too. A sample of the `main` method to run Duke is as follows
```java
 public static void main(String[] args){
     Duke duke = new Duke();
     duke.run();
}

```
**GLHF** :smile:
> It's something that was given to me, but I have a swimsuit calender hanging on the wall of my workplace. In short, I have 12 pictures of girls in bathing suits in my office. Nothing naughty. But anyways, every time the month changes, I have to flip the page to the next girl, and somehow, that makes me feel incredibly melancholy. 'Goody bye, September girl. I guess we'll never meet again.' That's how it makes me feel. And then I *flip*, and think, 'Wow, October's not bad either!", and that makes me think of how I'm already betraying the September girl... So, what I'm trying to say here is, that a year goes by really quickly. - Hirohiko Araki

jianh0ng committed:
Added Level-1

jianh0ng committed:
Added Level-2

jianh0ng committed:
Added Level-3

jianh0ng committed:
Added Level-4

jianh0ng committed:
Added Level-5

jianh0ng committed:
Added Level-6

jianh0ng committed:
Added A-TextUiTesting

jianh0ng committed:
Added javadoc comments

jianh0ng committed:
Add Level-7 in branch-Level-7

jianh0ng committed:
Add Level-8 in branch-Level-8

jianh0ng committed:
Merge branch 'branch-Level-7'
Add level-7 to the master

jianh0ng committed:
Merge branch 'branch-Level-8'
Merge branch-Level-8 into master

jianh0ng committed:
Add A-MoreOOP

jianh0ng committed:
Add A-Packages

jianh0ng committed:
Add A-JUnit

jianh0ng committed:
Add A-Jar

jianh0ng committed:
Add A-JavaDoc in branch-A-JavaDoc

jianh0ng committed:
Add A-CodingStandard in branch-A-CodingStandard

jianh0ng committed:
Add Level-9 in branch-Level-9

jianh0ng committed:
Merge branch 'branch-A-JavaDoc'
Add JavaDoc comments

jianh0ng committed:
Merge branch 'branch-A-CodingStandard'
Add A-CodingStandard

jianh0ng committed:
Merge branch 'branch-Level-9'
Add Level-9

jianh0ng committed:
Add A-Gradle in branch-A-Gradle

jianh0ng committed:
Merge branch 'branch-A-Gradle'
Add Gradle Support

jianh0ng committed:
Add A-CheckStyle to branch-A-CheckStyle

jianh0ng committed:
Merge branch 'branch-A-CheckStyle'
Add A-CheckStyle

jianh0ng committed:
Add Level-10 to branch-Level-10

jianh0ng committed:
Merge branch 'branch-Level-10'
Add Level-10

Droffilc13 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,60 @@
+
+        String remainder = "";
+        int separator = 0;
+        if (split.length > 1) {
```
I like how the Parser code is very readable as there is no deep nesting. 😄

meixuanjin replied:
There seem to be quite a few conditions here, maybe they could be abstracted out?

Droffilc13 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,60 @@
+public class Parser {
+
+    /**
+     * Returns the result of parsing the user input in an array of string where index 0 is command, index 1 is
```
I like that this Javadoc comment explains clearly the use of arbitrary integers 0, 1, 2 in the method body.

Droffilc13 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,60 @@
+            result[1] = remainder;
+        } else {
+            result[1] = remainder.substring(0, separator - 1);
+            result[2] = remainder.substring(separator + 4);
```
Perhaps a named constant instead of ```4```?

Droffilc13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+    /**
+     * ArrayList to store Tasks.
+     */
+    private ArrayList<Task> al;
```
Perhaps ```al``` could be renamed as ```tasks```?

Droffilc13 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,97 @@
+     * Echoes the given command.
+     * @param t Command to be given.
+     */
+    public void echo(String t) {
```
Any reason why you decided to keep method ```echo``` instead of deleting it?

meixuanjin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+        this.ui.showGreetMessage();
+        this.storage.load();
+        String userInput;
+        boolean isExit = false;
```
This boolean can be renamed to follow the coding standards more closely, for example hasExited.

meixuanjin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    public String getResponse(String input) {
```
Seems to be missing javadocs for this method?

meixuanjin commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,31 @@
+        super(desc[1], Boolean.valueOf(desc[3]));
+    }
+
+    public String toWrite() {
```
Missing javadocs for this method

meixuanjin reviewed (commented):
Everything looks good generally, and is very readable. The parts I commented were very tiny details but other than that, looks good!

Droffilc13 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+     * @param items Parsed delete command from user.
+     * @throws DukeException Exception for wrong user inputs.
+     */
+    public String deleteTask(String[] items) throws DukeException {
```
Any reason why you use an if-else block instead of just an if-block for the guard clause?

Droffilc13 reviewed (commented):
Overall, the code is easy to read, well-encapsulated and logic flows nicely. There are a few parts where I suggested changes, but all are very minor details for your consideration.

jianh0ng committed:
Fix fir JAR file to run properly

Edited build.gradle to launch the JavaFX GUI instead of text based.
Removed several redundant methods and method calls left from text GUI.

jianh0ng committed:
Add asserts to Duke and TaskList in branch-A-Assertions

jianh0ng committed:
Improve Code Quality

Code was unclear in some places, hence they have been changed/commented
to improve readability.

jianh0ng committed:
Merge pull request #1 from jianh0ng/branch-A-Assertions

Add asserts to Duke and TaskList in branch-A-Assertions

jianh0ng committed:
Merge branch 'master' into branch-A-CodeQuality

Resolved merge conflict

jianh0ng committed:
Merge pull request #2 from jianh0ng/branch-A-CodeQuality

Improve Code Quality

jianh0ng committed:
Add D-Places to branch-A-Extension

jianh0ng committed:
Merge branch 'branch-A-Extension'

Add D-Places functionality

jianh0ng committed:
Fix some parsing bugs and add delete for places

jianh0ng committed:
Add A-BetterGui

jianh0ng committed:
Set theme jekyll-theme-slate

jianh0ng committed:
Add Ui.png

jianh0ng committed:
Merge branch 'master' of github.com:jianh0ng/ip
Merging setup of github page with addition of Ui.png

jianh0ng committed:
Update README.md with user guide
